src/flint/flintlib/fq_default.pxd

```diff
@@ -0,0 +1,110 @@
+from flint.flintlib.fmpz_poly cimport fmpz_poly_t
+from flint.flintlib.fmpz cimport fmpz_t
+from flint.flintlib.flint cimport slong, flint_rand_t, ulong, mp_limb_t
+
+from flint.flintlib.nmod cimport nmod_t
+from flint.flintlib.nmod_poly cimport nmod_poly_t
+from flint.flintlib.fmpz_mod cimport fmpz_mod_ctx_t
+from flint.flintlib.fmpz_mod_poly cimport fmpz_mod_poly_t
+
+from flint.flintlib.fq cimport fq_t, fq_ctx_t
+from flint.flintlib.fq_nmod cimport fq_nmod_t, fq_nmod_ctx_t
+from flint.flintlib.fq_zech cimport fq_zech_t, fq_zech_ctx_t
+
+cdef extern from "flint/fq_default.h":
+    # TODO: how best to handle union
+    # https://stackoverflow.com/questions/12452210/cython-nesting-a-union-within-a-struct
+    ctypedef struct fq_default_struct:
+        fq_t fq
+        fq_nmod_t fq_nmod
+        fq_zech_t fq_zech
+        ulong nmod
+        fmpz_t fmpz_mod
+    ctypedef fq_default_struct fq_default_t[1]
```

Comment:
I'm not sure it matters but I think this should be `cdef union fq_default_struct`. The linked SO question refers to nested anonymous unions but this is just a union.

Reply:
I can modify this :)

Reply:
Tests work locally for me, will push the change for CI